### PR TITLE
Saas 13.5 layout designer preview improvements

### PR DESCRIPTION
**Pull request by SimonGenin**
The document layout preview is a complete different simplified template
with its own css that replicates at best the different styles.
It does not have the external layout features and lack of fidelity.

The new preview actually use the real documents templates and put the
result in an iframe. It now has a high fidelity, though not perfect.

The goal is for a better onboarding, where clients see easely how
documents will look if they had an app to generate them. Of course, the
data on the document is a false invoice.

Refactor all this from base to web.

Task ID 2304177

### CODE REVIEW

**Comment by SimonGenin** [comment]
@kebeclibre

**Comment by SimonGenin** [comment]
Migration : https://github.com/odoo/upgrade/pull/1675

File: addons/web/models/base_document_layout.py [hunk 1]
```diff
@@ -33,9 +45,27 @@ class BaseDocumentLayout(models.TransientModel):
...
+                          sanitize_form=False,
+                          strip_style=False,
+                          strip_classes=False)
+
+    # Those following fields are required as a company to create invoice report
+    partner_id = fields.Many2one(related='company_id.partner_id', readonly=False)
+    phone = fields.Char(related='company_id.phone', readonly=False)
+    email = fields.Char(related='company_id.email', readonly=False)
+    website = fields.Char(related='company_id.website', readonly=False)
+    vat = fields.Char(related='company_id.vat', readonly=False)
```
Comment (kebeclibre) [inline addons/web/models/base_document_layout.py:64]: readonly=True :D

**Comment by kebeclibre** [comment]
migration https://github.com/odoo/upgrade/pull/1686

File: addons/web/models/base_document_layout.py (hunk 1 above)
Comment (SimonGenin) [reply addons/web/models/base_document_layout.py:64]: Champion ici

Comment (SimonGenin) [reply addons/web/models/base_document_layout.py:64]: fixed

**Comment by KangOl** [comment]
Please remove the overwatch image. It's subject to copyrights. You can ask your designer team for some original logo for this purpose.

**Comment by kebeclibre** [comment]
robodoo r+

**Comment by kebeclibre** [comment]
@KangOl 
This is the feature PR of the upgrade PR you merged. What should I do here ? by pass the mergebot and do a classic git push ?

**Comment by KangOl** [comment]
@Xavier-Do can the runbot be tolerant?

**Comment by Xavier-Do** [comment]
Exceptions added to merge this pr in stable: 

`view:web.view_base_document_layout,view:web.report_invoice_wizard_preview,view:web.html_preview_container,web.report_preview_layout,field:base.document.layout.country_id,field:base.document.layout.name,field:base.document.layout.vat,field:base.document.layout.website,field:base.document.layout.email,field:base.document.layout.phone,field:base.document.layout.partner_id,field:report.layout.sequence`

**Comment by Xavier-Do** [comment]
@KangOl Looks like it is not enough: we are in the `either it is a custom field marked as coming from a standard module due to odoo/odoo#49354.` I guess that I can move the exception on top of all field cases to fix this issue.

**Pull request closed without merging**